packages/mermaid/src/diagrams/xychart/xychartDetector.ts

```diff
@@ -0,0 +1,20 @@
+import type { DiagramDetector, ExternalDiagramDefinition } from '../../diagram-api/types.js';
+
+const id = 'xychart';
+
+const detector: DiagramDetector = (txt) => {
+  return txt.match(/^\s*xychart/i) !== null;
```

Comment:
```suggestion
  return /^\s*xychart-beta/.test(txt);
```

I've two question about this line:

1. Shouldn't we add `-beta` suffix here?
2. Is there a reason to add `i` flag?

Reply:
Intentionally left the `-beta` so that it will work without beta but in the documentation and example we have added beta
i flag is just to have case insensitive

Reply:
> Intentionally left the `-beta` so that it will work without beta but in the documentation and example we have added beta

Will it be able to detect it if it doesn't have `-beta` suffix?
The main goal of `-beta` suffix is to indicate that the diagram grammar hasn't been finalized, allowing the keyword without would indicate that we're finalized it and we're just supporting `-beta` as backward compatibility.

> i flag is just to have case insensitive

That's my question, why do we need that? All other diagrams have case sensitive keyword.

Reply:
I have changed those.
for the case insensitive part, I thought since the language is case insensitive why not the diagram name?